File: app/src/main/java/tk/lorddarthart/vkauthtestapp/MainActivity.java
package tk.lorddarthart.vkauthtestapp;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.vk.sdk.VKSdk;

public class MainActivity extends AppCompatActivity {
    Fragment fr1;
    FragmentTransaction transaction;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        VKSdk.initialize(getApplicationContext());
        if (true) {
            fr1 = new FragmentFriendslist();
            transaction = getSupportFragmentManager().beginTransaction();
            transaction.replace(R.id.mainFragment, fr1);
            transaction.commit();
        } else {
            fr1 = new FragmentAuth();
            transaction = getSupportFragmentManager().beginTransaction();
            transaction.replace(R.id.mainFragment, fr1);
            transaction.commit();
        }
    }
}

File: app/src/main/java/tk/lorddarthart/vkauthtestapp/app/view/activity/MainActivity.kt
package tk.lorddarthart.vkauthtestapp.app.view.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import tk.lorddarthart.vkauthtestapp.R
import tk.lorddarthart.vkauthtestapp.util.marker.IOnBackPressable
import tk.lorddarthart.vkauthtestapp.util.helper.isOnBackPressable

class MainActivity : AppCompatActivity() {
    val currentBaseFragment: Fragment?
        get() {
            var fragment: Fragment? = null
            if (supportFragmentManager.findFragmentById(R.id.container_base) != null) {
                fragment = supportFragmentManager.findFragmentById(R.id.container_base)
            }
            return fragment
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_activity)
    }

    override fun onBackPressed() {
        if (currentBaseFragment?.isOnBackPressable == true) {
            (currentBaseFragment as IOnBackPressable).onBackPressed()
        } else {
            finishAffinity()
        }
    }
}